### DISCUSSION TITLE
Easier inserts with many columns

### DISCUSSION THREAD

**Discussion by alecbz** [sqlc-dev/sqlc#587] [Issue Triage] [answered]
Suppose I have some table with a lot of columns, and we get bulk API requests with those fields specified. Right now, as far as I can tell, I'd need to have a query like:

```sql
-- name: InsertThing :one
INSERT INTO members (a, lot, of, fields, that, this, type, of, data, has)
VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10)
RETURNING *;
```

This isn't the end of the world, but I'm wondering if it'd be possible to support something like:

```sql
-- name: InsertThing :one
INSERT INTO members (a, lot, of, fields, that, this, type, of, data, has)
VALUES ($*)
RETURNING *;
```

?

**Comment by landbed** [comment]
how about this?

```
-- name: InsertThing :one
INSERT INTO members ($*)
VALUES ($*)
RETURNING *;
```

and generating codes like:

https://github.com/jirfag/go-queryset/blob/master/examples/comparison/gorm4/autogenerated_gorm4.go#L547

https://github.com/jirfag/go-queryset/blob/master/examples/comparison/gorm4/autogenerated_gorm4.go#L566

to updating optional fields, that is dynamic and type-safe.

actually UPDATE, SELECT, INSERT, WHERE etc are same.

ping: https://github.com/kyleconroy/sqlc/issues/580

**Comment by alecbz** [comment]
@landbed That might be getting a bit too magic. It's usually not the case that _every_ column should be specified in an insert statement (e.g., an auto incrementing id), and I imagine sqlc wants to stay out of the business of guessing which columns are and aren't necessary in an insert statement.

**Comment by kyleconroy** [comment, marked as answer]
> and I imagine sqlc wants to stay out of the business of guessing which columns are and aren't necessary in an insert statement.

You are correct that I'd like sqlc to avoid guessing which columns should be included and which ones shouldn't. 

When adding SQL enhancements, I like to add them as functions in the sqlc namespace. So maybe it should look like this?


```sql
-- name: InsertThing :one
INSERT INTO members (a, lot, of, fields, that, this, type, of, data, has)
VALUES (sqlc.args())
RETURNING *;
```

or 

```sql
-- name: InsertThing :one
INSERT INTO members (a, lot, of, fields, that, this, type, of, data, has)
VALUES sqlc.values()
RETURNING *;
```

**Comment by alecbz** [comment]
Both of those LGTM!
